src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonIgnoreMetadataNamesDictionaryKeyFilter.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json.Serialization;
+
+/// <summary>
+/// Determines dictionary key filter to ignore metadata keys starting with $, such as `$schema`.
+/// </summary>
+internal sealed class JsonIgnoreMetadataNamesDictionaryKeyFilter : JsonDictionaryKeyFilter
+{
+
+    /// <summary>
+    /// Ignores any metadata keys starting with $, such as `$schema`.
```

Comment:
Rather than ignoring _every_ key starting with `$`, I would recommend only checking for metadata property names that have built-in support in JSON:

https://github.com/dotnet/runtime/blob/5ca682626651957080b6b4c42cb2ee1c5e743eb3/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs#L19-L22